pkg/streamingpromql/types/fpoint_ring_buffer.go

```diff
@@ -169,12 +145,19 @@ func (b *FPointRingBuffer) Release() {
 // s will be modified in place when the buffer is modified, and callers should not modify s after passing it off to the ring buffer via Use.
 // s will be returned to the pool when Close is called, Use is called again, or the buffer needs to expand, so callers
 // should not return s to the pool themselves.
+// s must have a capacity that is a power of two.
 func (b *FPointRingBuffer) Use(s []promql.FPoint) {
+	if !isPowerOfTwo(cap(s)) {
```

Comment:
With my above suggestion, this is perhaps a little more risky to leave unenforced, so may be worth keeping here. However, in theory, so long as everything has come from the bucketed pool then we should always have the right power.

Reply:
This check is cheap, and tracking down an issue related to this could be quite painful, so I'd prefer to keep it for now.